app/src/main/java/clarkson/ee408/tictactoev4/MainActivity.java

```diff
@@ -26,20 +26,83 @@ public class MainActivity extends AppCompatActivity {
     private Button[][] buttons;
     private TextView status;
     private Gson gson;
+    private boolean shouldRequestMove = false;
+    private SocketClient socketClient;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         this.tttGame = new TicTacToe(STARTING_PLAYER_NUMBER);
         this.gson = new GsonBuilder().serializeNulls().create();
+        socketClient = SocketClient.getInstance();
 
         buildGuiByCode();
+        updateTurnStatus();
 
         Handler handler = new Handler();
         GameMoveTaskRunnable runnable = new GameMoveTaskRunnable(this, handler);
         handler.post(runnable);
     }
 
+    private void requestMove() {
```

Comment:
Make this public and add javadoc

```suggestion
    /**
     * Sends a request to the server to ask for a game move made by the other player.
     */
    public void requestMove() {
```

Reply:
Yes. You are right.